### PR TITLE
Fix ValuesToRange ConditionOptimizer

### PR DESCRIPTION
**Pull request by sstok**
The optimizer failed to optimize other connected value-lists and only optimized the first one.

### CODE REVIEW

File: tests/ConditionOptimizer/ValuesToRangeTest.php [hunk 1]
```diff
@@ -43,6 +43,16 @@ public function it_converts_single_proceeding_values_to_ranges()
                 ->addSimpleValue(5)
                 ->addSimpleValue(10)
                 ->addSimpleValue(7)
+                ->addSimpleValue(12)
+                ->addSimpleValue(13)
+                ->addSimpleValue(14)
+                ->addSimpleValue(15)
+                // edge-case where first non-previous increment it start of new increment
```
Comment (sstok) [inline tests/ConditionOptimizer/ValuesToRangeTest.php]: It's late I will fix this later... 😴

**Comment by sstok** [comment]
Again. use the the `-b` option when checking out a branch 💀
